C-Plus-Plus/dp/Zero_One_Knapsack.cpp

```diff
@@ -19,24 +25,36 @@ void input() {
   for (int i = 0; i <= n; ++i)
     table[i] = new int[k + 1];
 
-  for (int i = 0; i <= n; i++) {
-    for (int j = 0; j <= k; j++)
-      table[i][j] = -1;
-  }
-}
-
-//finding maximum profit in iterative way
-void knapsack() {
-  for (int j = 0; j <= k; j++)
-    table[0][j] = 0;
-  for (int i = 0; i <= n; i++)
-    table[i][0] = 0;
-  for (int i = 1; i <= n; i++) {
-    for (int j = 1; j <= k; j++) {
-      if (j < weight[i]) //when size of knapsack is less than weight of item i
-        table[i][j] = table[i - 1][j];
-      else
-        table[i][j] = max(table[i - 1][j], table[i - 1][j - weight[i]] + profit[i]);
+//Taking input
+void input()			
+{
+	cin>>n;
+	cin>>k;
+	profit=new int[n];
+	weight=new int[n];
+	for(int i=1; i<=n; i++)
+    {   
+		cin>>profit[i];
+		cin>>weight[i];		
+	}
+ }
+int knapsack(int n,int k)
```

Comment:
Try to format your code according to [C++ style guide](https://google.github.io/styleguide/cppguide.html). There should be proper space between operator and operand, a space after comma ```,```, etc. Go through the standard C++ style guide once, it will help you to write cleaner and better code in future. Happy coding.